### ISSUE TITLE
tflite model compiled from h5 outputs just nan

### ISSUE THREAD

**Issue by ptrem** [tensorflow/tensorflow#42927]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10
- TensorFlow installed from (source or binary): pip
- TensorFlow version (or github SHA if from source): 2.3.1

Hey everybody,
i am still relatively inexperienced with tensorflow lite and would like to convert a model with custom training function into a tflite model in order to run it on a google coral accelerator if possible. I am training the model on a NVIDIA Tesla V100 and would like to convert the saved model/graph (.pb, h5) into a tflite.
My model is almost identical to the one in cycleGan Tutorial. 
After the conversion into the tflite format I get only "NAN" as model output.
I have also tried to convert the model from the tutorial into tflite and the same error occurs.
I have also tried different conversion types, Tensorflow versions etc. and nothing worked.
Are there possibly any functions in it which tflite does not support?

Thanks a lot in advance.

**Comment by ravikyram** [comment]
@ptrem 

Can you please share colab link or simple standalone code with supporting files to reproduce the issue in our environment. It helps us in localizing the issue faster. Thanks!

**Comment by ptrem** [comment]
The Notebook tf_cyclehorse.ipynb is used to train the model and to create the .h5 file. 
To convert the .h5 file to a .tflite file I use a script similar to conv_tflite.py. 
(to run the file you use the console comand similar to: python conv_tflite.py --model model.h5 --output model.tflite)
 tflite_test.ipynb is a simple notebook to test the return of the tflite model. 
Thereby the image horse.jpg and the model model.tflite is read in. The output of the errors is then reproduced.

Follow the google drive linke:
I've attached all the notebooks, scripts, models and images. Almost everything you need.
https://drive.google.com/drive/folders/1U-YZX0k48QaEK5ZBQviHsNdV9yzCfFqZ?usp=sharing

**Comment by ptrem** [comment]
Is there no solution for my problem?

**Comment by ptrem** [comment]
I realized that the error occurs, when using the tf.data.dataset API. Is there any relation?

**Comment by ptrem** [comment]
It seemed to be a problem of the tf.keras.layers.BatchNormalization().
If this issue occurs just set the attribute fused=False and it should work fine.
